broker.go

```diff
@@ -1337,17 +1337,19 @@ consumeRetryLoop:
 					// In order to keep API for Consumer
 					// here we repack Records to Messages
 
-					messages := make([]*proto.Message, 0, len(part.RecordBatch.Records))
-					for _, r := range part.RecordBatch.Records {
-						m := &proto.Message{
-							Key:       r.Key,
-							Value:     r.Value,
-							Offset:    part.RecordBatch.FirstOffset + r.OffsetDelta,
-							Topic:     topic.Name,
-							Partition: part.ID,
-							TipOffset: part.TipOffset,
+					var messages []*proto.Message
```

Comment:
I'm going to come back and add code to count the messages so that we can pre-allocate this slice, but there is no test coverage of this code, so I will add that first...